src/SourceFiles/ConfettiWeapon.cpp

```diff
@@ -41,12 +42,32 @@ void ConfettiWeapon::setActive(bool a, b2Vec2 pos)
 		if (currentHand_ != nullptr) UnPickObject();
 		viewer_->setFrame(0);
 		used = false;
-		cout << "tesst";
 		colWeapon_->setLinearVelocity(b2Vec2(0, 0));
 		colWeapon_->setAngularVelocity(0);
+		currentFrame_ = 0;
 	}
 	entity_->setActive(a);
 	viewer_->setDrawable(a);
 	colWeapon_->getBody()->SetEnabled(a);
 	colWeapon_->getBody()->SetTransform(pos, 0);
 }
+
+void ConfettiWeapon::update()
+{
+	MeleeWeapon::update();
+	if (used) {
+		currentFrame_++;
+		if (currentFrame_ == animationDuration_) {
+			if (currentHand_) currentHand_->setFrame(2, weaponType_);
```

Comment:
```suggestion
			if (currentHand_ != nullptr) currentHand_->setFrame(2, weaponType_);
```
Os recuerdo que si sólo ponéis el puntero como un booleano estáis comparando otra cosa